core/state_processor.go

```diff
@@ -713,6 +724,79 @@ func (p *StateProcessor) Process(block *types.WorkObject, batch ethdb.Batch) (ty
 				timeDelta := time.Since(startTimeEtx)
 				timeCoinbase += timeDelta
 				continue
+			} else if etx.EtxType() == types.ConversionRevertType {
+				// checking to make sure that the conversions that have to be
+				// refunded and processed before processing etxs from other
+				// zones
+				to := tx.To()
+				var sender common.Address
+				if to.IsInQuaiLedgerScope() {
+					sender = common.BytesToAddress(tx.Data()[2:22], p.hc.NodeLocation())
+				} else {
+					sender = tx.ETXSender()
+				}
+
+				// If the sender is in Quai ledger scope and to is in Qi,
+				// the reverted transaction should add the balance back to
+				// the original sender (Quai ledger)
+				if sender.IsInQuaiLedgerScope() && to.IsInQiLedgerScope() {
+					senderInternal, err := sender.InternalAddress()
+					if err != nil {
+						return nil, nil, nil, nil, 0, 0, 0, nil, nil, err
+					}
+					statedb.AddBalance(senderInternal, etx.Value())
```

Comment:
A refund for Quai->Qi isn't locked but a refund for Qi->Quai is locked for 2 weeks? Is that intentional?